Option to save the downsampled point cloud

Luc1100 opened the pull request:
Option to save the downsampled point cloud to a .txt file. Useful for importing points to Dynamo to do subtomogram averaging along a membrane using a surface model.

maurerv commented:
Hi @Luc1100,

Thank you very much for your contribution. I managed to have a look at your code and was wondering whether you could not achieve a similar behavior by simply selecting all clusters and then using the "Save Selected TXT" button from the "Save" section. The text file written to disk would, however, not be downsampled.

On another note, when sampling from colabseg's parameterizations, we use triangular meshes to ensure a particular sampling rate of the mesh, e.g., a minimum distance of 15 Angstrom separates all points. Would a similar minimum distance for writing point clouds to disk be beneficial for your use case?

Luc1100 commented:
Hi @maurerv,

The issue I've run into with just using the "Save Selected TXT" button is that since it is the entire point cloud, the membrane ends up having some thickness to it. Additionally this point cloud has way more points than I need to define the general shape of the membrane (almost 200,000 points for what I am segmenting). Also, bits from other membranes that do not show up in the editor viewer end up saving as well which makes the clean up a little hard because you have to guess which part of the membrane the points are being pulled from.

For what I am doing, I just need the general shape of the membrane which the downsampled cloud in the viewer is perfect for because it has no thickness and the points are nicely distributed, so the button that I added saves only these coordinates. This also significantly reduces the number of points from ~200,000 to ~1,000. I even had success generating a nice average just using these points without creating any special models in Dynamo. 

I do think it would be great if instead of only saving the point cloud that is displayed like I've implemented, you could enter some minimum separation for all the points and be able to save any arbitrary down sampled cloud. I just wanted a quick and dirty way to save what is displayed for the moment.

maurerv commented:
Okay, then that is something to put on the list moving forward.

Thanks a lot for the contribution again, I will go ahead and merge your PR.
